study/src/test/java/di/stage4/annotations/ClassPathScanner.java

```diff
@@ -2,9 +2,13 @@
 
 import java.util.Set;
 
+import org.reflections.Reflections;
+import org.reflections.scanners.Scanners;
+
 public class ClassPathScanner {
 
     public static Set<Class<?>> getAllClassesInPackage(final String packageName) {
-        return null;
+        final var reflections = new Reflections(packageName, Scanners.SubTypes.filterResultsBy(filter -> true));
```

Comment:
아까 냥인이 물어보셨던 거에 대한 답변입니다!!이미 알아내셨다면 다행이지만 그래도 저도 겸사겸사 남겨봅니다ㅎㅎ
여기서 `filterResultBy(filter -> true)`를 붙이지 않을 경우 오류가 나는 이유는 `Scanners.SubTypes`의 기본 필터링 설정에서 Object.class는 제외한다는 설정이 있어서 그렇습니다!!
해당 클래스를 확인해보면 이런 설명이 있더라구요.
>scan type superclasses and interfaces Note that Object class is excluded by default, in order to reduce store size. Use filterResultsBy(Predicate) to change, for example SubTypes. filterResultsBy(c -> true)
<img width="731" alt="스크린샷 2024-10-02 17 53 35" src="https://github.com/user-attachments/assets/d5616818-2ef4-4095-9308-ffcbbbf5d44b">

그래서 `filter->true`를 하지 않으면 오류가 발생했던 것 같습니다😁

Reply:
오 Scanners를 설정해 줘도 `filter->true`를 안 하면 오류가 나는군요.

> Scanners must be configured in order to be queried, otherwise an empty result is returned.

저는 위 문장만 보고 Scanners를 꼭 설정해 줘야 하는 거구나~ 정도로만 파악했는데, 
초롱은 더 세세한 내용까지 파악하고 계셨네요. 👍

알려주셔서 정말로 감사합니다. 😸